packages/mds-cache/unflatteners.ts

```diff
@@ -75,82 +79,75 @@ function parseAllDeviceStates(allDeviceStates: StringifiedAllDeviceStates): { [v
-            : null,
-          service_area_id: tripEvent.service_area_id ? tripEvent.service_area_id : null
-        })
+    const result: TripEvent[] = []
+    const tripEvents: StringifiedTripEvent[] = JSON.parse(tripEventsStr)
+    tripEvents.map(tripEvent => {
+      result.push({
+        vehicle_type: tripEvent.vehicle_type as VEHICLE_TYPE,
+        timestamp: Number(tripEvent.timestamp),
+        event_type: tripEvent.event_type as VEHICLE_EVENT,
+        event_type_reason: tripEvent.event_type_reason ? (tripEvent.event_type_reason as VEHICLE_REASON) : null,
+        annotation_version: Number(tripEvent.annotation_version),
+        annotation: tripEvent.annotation
+          ? {
+              in_bound: JSON.parse(tripEvent.annotation.in_bound),
+              areas: tripEvent.annotation.areas
+            }
+          : null,
+        gps: tripEvent.gps
+          ? {
+              lat: Number(tripEvent.gps.lat),
+              lng: Number(tripEvent.gps.lng),
+              altitude: tripEvent.gps.altitude ? Number(tripEvent.gps.altitude) : null,
+              heading: tripEvent.gps.heading ? Number(tripEvent.gps.heading) : null,
+              speed: tripEvent.gps.speed ? Number(tripEvent.gps.speed) : null,
+              accuracy: tripEvent.gps.accuracy ? Number(tripEvent.gps.accuracy) : null
+            }
+          : null,
+        service_area_id: tripEvent.service_area_id ? tripEvent.service_area_id : null
       })
     })
-    return trips
+    return result
   } catch (err) {
-    throw new ParseError(`unable to parse tripsEvents: ${tripsEventsStr}: ${err}`)
+    throw new ParseError(`unable to parse tripsEvents: ${tripEventsStr}: ${err}`)
   }
 }
 
-function parseTripsTelemetry(tripsTelemetryStr: StringifiedTripsTelemetry): TripsTelemetry {
+function parseTripTelemetry(tripTelemetryStr: StringifiedTripTelemetries): TripTelemetry[] {
   try {
-    const trips: TripsTelemetry = {}
-    const tripsTelemetry: StringifiedTripsTelemetry = JSON.parse(String(tripsTelemetryStr))
-    Object.keys(tripsTelemetry).map(trip_id => {
-      trips[trip_id] = []
-      tripsTelemetry[trip_id].map(tripTelemetry => {
-        trips[trip_id].push({
-          timestamp: Number(tripTelemetry.timestamp),
-          latitude: tripTelemetry.latitude ? Number(tripTelemetry.latitude) : null,
-          longitude: tripTelemetry.longitude ? Number(tripTelemetry.longitude) : null,
-          annotation_version: Number(tripTelemetry.annotation_version),
-          annotation: tripTelemetry.annotation
-            ? {
-                in_bound: JSON.parse(tripTelemetry.annotation.in_bound),
-                areas: tripTelemetry.annotation.areas
-              }
-            : null,
-          service_area_id: tripTelemetry.service_area_id ? tripTelemetry.service_area_id : null
-        })
+    const result: TripTelemetry[] = []
+    const tripTelemetry: StringifiedTripTelemetry[] = JSON.parse(tripTelemetryStr)
+    tripTelemetry.map(telemetry => {
```

Comment:
Same as above comment on not pushing and rather just doing `const result = tripTelemetry.map (...)`